CORDA-3953 - Re-enabled the IRSDemoTest and added some logging

filipesoliveira opened the pull request:
**Bug Reported**
The IRS Demo test was flaky before being disabled but the test should be re-enabled once it is stabled.

**Bug Summary**
It was not possible to reproduce the issue after running the test several times. The test has been enabled again and more logging was added to the IRS Demo test so we can hopefully have a better understanding of what is causing the issue next time the test fails

**Note**
- Keep in mind that this PR does not fix the issue. It merely adds more logging to the test so we can have more data about the test next time it fails.
- The logging prints the elapsed time each function call takes and the overall time the test takes to execute if successful.

filipesoliveira committed:
Re-enabled the IRSDemoTest and added some logging to gather more data about the issue

filipesoliveira committed:
Updated a variable name to avoid variable shadowing.

rnicoll reviewed (changes requested):
So far seems to be passing the IRS demo test, although another test glitched. I think fixes introduced elsewhere have improved the state of this, and I'd like to get this merged.

However, I think lets not reinvent the wheel on formatting - what you have is definitely a nicely formatted string, but I'm not sure it's worth having extra code to maintain. Consider instead using https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html to track the interval, and it has a `.toString()` method that produces a standardised string.

filipesoliveira committed:
Refactored the code that was added in the previous two commits.

Changes summary:
1) The Timer class was added to group all the functions that were previously added for the timer logic
2) The class Duration is now used to print the elapsed time in a standardise format.

filipesoliveira committed:
Updated the function calculateElapsedTime so it uses the class property instead of using an argument.

filipesoliveira commented:
> So far seems to be passing the IRS demo test, although another test glitched. I think fixes introduced elsewhere have improved the state of this, and I'd like to get this merged.
> 
> However, I think lets not reinvent the wheel on formatting - what you have is definitely a nicely formatted string, but I'm not sure it's worth having extra code to maintain. Consider instead using https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html to track the interval, and it has a `.toString()` method that produces a standardised string.

The code now uses the Duration class to produce a standardised string for the elapsed time. I also created a Timer class to group the functions I have previously added. This should also help minimising duplication.
